src/components/Settings/SettingsTab.tsx

```diff
@@ -53,12 +65,28 @@ export const SettingsTab = ({
   isChatDisabled,
   setIsChatDisabled,
   clearChat,
+  roomTitle,
+  roomDescription,
+  roomTitleColor,
 }: SettingsTabProps) => {
   const [updateTS, setUpdateTS] = useState(0);
   const [permModalOpen, setPermModalOpen] = useState(false);
   const [validVanity, setValidVanity] = useState(true);
   const [validVanityLoading, setValidVanityLoading] = useState(false);
   const [adminSettingsChanged, setAdminSettingsChanged] = useState(false);
+  const [roomTitleInput, setRoomTitleInput] = useState<string | undefined>('');
+  const [roomDescriptionInput, setRoomDescriptionInput] = useState<string | undefined>(
+    ''
+  );
+  const [roomTitleColorInput, setRoomTitleColorInput] = useState<string | undefined>(
+    ''
+  );
+
+  useEffect(() => {
```

Comment:
curious why we need a new effect for these

Reply:
Initializing the input fields with those values, these props are not immediately available when this component is mounted.

Reply:
How are we handling this for the other settings fields? should work the same for those right?

Reply:
Hmm, I might be missing something here but I think what's so confusing about this is that we have two different states.

roomTitle which is in the parent component and updates the top bar and roomTitleInput which updates the input field text and the roomTitle state if admin settings are saved. On initial page load we need to set roomTitleInput equal to roomTitle which is undefined when SettingsTab is mounted initially. The other settings field text values are all Props which will force a re-render of SettingsTab if they go from initially undefined to some text.

If we used a single state to update both the top bar and the input field the person typing in a new title would see the top bar update as he types and might think that the changes have already been applied.

Reply:
I agree we should have two different states to avoid confusion. But do we need to sync up the parent component state with the input state in the SettingsTab? The local user sees the changes they type, and we emit an update to the entire room when admin settings change for everyone else right?

The parent component can update on the emit event